spring-cloud-gcp-logging/src/main/java/org/springframework/cloud/gcp/logging/CompositeTraceIdExtractor.java

```diff
@@ -0,0 +1,57 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.logging;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.common.collect.ImmutableList;
+
+import org.springframework.util.Assert;
+
+/**
+ * Checks HTTP request for multiple headers that might contain a trace ID, and provides
```

Comment:
That's not true. It just delegates to the provided set of `TraceIdExtractor`s and that is already their responsibility to check HTTP headers or get the value from the session or some other place.

Please, fix the JavaDoc according the current state.

Reply:
This isn't actually outdated. What I have there is true, but just less detailed than your description. I'll make the change.

Reply:
??? The `TraceIdExtractor` strategy doesn't state that an implementation should always look only into the headers. How about if I'm going to resolve it from the request params, or from body, or just from session attribute? Or just from static value from testing purpose. Can I? So, why them to restrict composite purpose just to headers by its JavaDocs or just mislead that way?

Sorry for the confuse but the feature is really confusing...